src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -7960,19 +8040,8 @@ cpe *wm_vuldet_generate_cpe(const char *part, const char *vendor, const char *pr
     return new_cpe;
 }
 
-int wm_vuldet_run_scan(wm_vuldet_t *vuldet) {
-    int result = OS_INVALID;
-    mtinfo(WM_VULNDETECTOR_LOGTAG, VU_START_SCAN);
-
-    if (wm_vuldet_nvd_empty() == 0) {
-        mterror(WM_VULNDETECTOR_LOGTAG, VU_NVD_EMPTY);
-        return result;
-    }
-
-    result = wm_vuldet_check_agent_vulnerabilities(vuldet);
-
-    scan_agent *agent;
-    for (agent = vuldet->scan_agents; agent;) {
+void wm_vuldet_clean_all_scan_agent_structures(scan_agent* agent) {
```

Comment:
Great! It would be good to use it also in wm_vuldet_destroy() function.

Reply:
You are right !!